src/components/InputV2/PrivateKey.tsx

```diff
@@ -35,7 +35,12 @@ export const PrivateKeyInput = ({
 
   const onChangeBase = (newValue: string) => {
     setHasChanged(true)
-    onChange(newValue.trim())
+    onChange(
+      isHex
+        ? // Ensure hex is lowercase and strip the 0x prefix
+          newValue.trim().toLocaleLowerCase().replace(/^0x/, '')
+        : newValue.trim()
+    )
```

Comment:
The input validation will fail when a hex is uppercase. Also it will fail when adding a `0x` prefix for hex numbers. This forces lowercase and agressivelyt removes `0x` from the beginning.